source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2475 @@
+[Why are extraOptions and kmsProviders maps?](#why-are-extraoptions-and-kmsproviders-maps).
+
+<div id="extraoptions.cryptsharedlibpath">
+
+##### `extraOptions.cryptSharedLibPath`
+
+- Type: `undefined | string`
+- Default: `undefined`
+
+Allow the user to specify an absolute path to a [crypt_shared](#crypt_shared) dynamic library to load. Refer:
+
+- [Overriding the crypt_shared Library Path](#overriding-the-crypt_shared-library-path)
+- [Path Resolution Behavior](#path-resolution-behavior)
+- \[Enabling crypt_shared\](#Enabling crypt_shared)
+
+<div id="extraoptions.cryptsharedlibrequired">
+
+##### `extraOptions.cryptSharedLibRequired`
+
+- Type: `boolean`
+- Default: `false`
+
+If `true`, the driver MUST refuse to continue unless [crypt_shared](#crypt_shared) was loaded successfully.
+
+If, after initializing a `libmongocrypt_handle`, [crypt_shared](#crypt_shared) is detected to be unavailable AND
+[extraOptions.cryptSharedLibRequired](#extraoptions.cryptsharedlibrequired) is `true`, the driver MUST consider the
+`libmongocrypt_handle` to be invalid and return an error to the user. Refer:
+
+<div id="#enabling-crypt_shared">
+<div id="managing-mongocryptd">
+<div id="#detecting-crypt_shared-availability">
+
+#### encryptedFieldsMap
+
+`encryptedFieldsMap` maps a collection namespace to an `encryptedFields`.
+
+`encryptedFieldsMap` only applies to Queryable Encryption.
+
+If a collection is present on both the `encryptedFieldsMap` and `schemaMap`, [libmongocrypt](#libmongocrypt) will error
+on initialization. See `fle2-and-fle1-error`.
+
+If a collection has a set of encrypted fields, the behavior of `CreateCollection()` and `Collection.Drop()` is altered.
+An additional helper, `CreateEncryptedCollection()` has been defined as a convenience wrapper around
+`CreateCollection()`. See `fle2-createcollection-drop`.
+
+Automatic encryption in Queryable Encryption is configured with the `encryptedFields`.
+
+If a collection is not present on the `encryptedFields` a server-side collection `encryptedFields` may be used by
+[libmongocrypt](#libmongocrypt). Drivers MUST include the following in the documentation for MongoClient:
+
+> Supplying an `encryptedFieldsMap` provides more security than relying on an `encryptedFields` obtained from the
+> server. It protects against a malicious server advertising a false `encryptedFields`.
+
+#### bypassQueryAnalysis
+
+See [Why is bypassQueryAnalysis needed?](#why-is-bypassqueryanalysis-needed).
+
+### Queryable Encryption Create and Drop Collection Helpers
+
+A collection supporting Queryable Encryption requires an index and three additional collections.
+
+<div id="GetEncryptedFields">
+
+#### Collection `encryptedFields` Lookup (GetEncryptedFields)
+
+The convenience methods support the following lookup process for finding the `encryptedFields` associated with a
+collection.
+
+Assume an exposition-only function $GetEncryptedFields(opts, collName, dbName, askDb)$, where $opts$ is a set of
```

Comment:
I'll note that these render as [MathML](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions), which makes them hard to copy-paste (if that's a concern).

These terms were previously formatted with single backticks in the RST, so would it make sense to keep them as monospace text here?

Reply:
Done